samples/csharp_dotnetcore/14.nlp-with-dispatch/Startup.cs

```diff
@@ -186,7 +186,7 @@ private static BotServices InitBotServices(BotConfiguration config)
 
                             var app = new LuisApplication(luis.AppId, luis.AuthoringKey, luis.GetEndpoint());
                             var recognizer = new LuisRecognizer(app);
-                            luisServices.Add(luis.Name.Split("_").LastOrDefault(), recognizer);
```

Comment:
why?

Reply:
This is actually due to the differences in workflow when following the instructions. If you follow the instructions for _cloning_ the bot via `msbot clone`, you currently have to split on an underscore to get your actual LUIS application's name. If you follow the manual instructions, the bot's name isn't added to the LUIS application's name.

This prefixing of the bot's name to the LUIS name is problematic, so @daveta filed an [issue in botbuilder-tools](https://github.com/Microsoft/botbuilder-tools/issues/667) and is working on adding a flag to disable the prefixing.

After the fix is in we should update the READMEs in the samples to include this flag by default and then take out the `luis.Name.Split("_")` code across samples.